File: leetcode/algorithms/724_find_pivot_index/main.c
int sum(int* nums, int numsSize) {
    int num = 0;
    for (int i = 0; i < numsSize; i++) {
        num += nums[i];
    }
    return num;
}

int pivotIndex(int* nums, int numsSize) {
    int leftSum = 0;
    int rightSum = sum(nums, numsSize);

    for (int i = 0; i < numsSize; i++) {
        rightSum -= nums[i];

        if (leftSum == rightSum) {
            return i;
        }

        leftSum += nums[i];
    }

    return -1;
}


// Solution
int solution(int* nums, int numsSize) {
    // Initialize rightSum to store the sum of all the numbers
    // strictly to the index's right
    int rightSum = sum(nums, numsSize);

    // Initialize leftSum to store the sum of all the numbers
    // strictly to the index's left
    int leftSum = 0;

    // Traverse all elements through the loop
    for (int idx = 0; idx < numsSize; idx++) {
        // subtract current elements with from rightSum
        rightSum -= nums[idx];

        // If the sum of all the numbers strictly to the left
        // of the index is equal to the sum of all the numbers
        // strictly to the index's right
        if (leftSum == rightSum) {
            // Return the pivot index
            return idx;
        }

        // add current elements with leftSum
        leftSum += nums[idx];
    }

    // If there is no index that satisfies the conditions
    // in the problem statement
    return -1;
}

File: leetcode/algorithms/724_find_pivot_index/main.py
from typing import List


class FindPivotIndex:
    def pivotIndex(self, nums: List[int]) -> int:
        leftSum: int = 0
        rightSum: int = sum(nums)

        for i in range(len(nums)):
            rightSum -= nums[i]

            if leftSum == rightSum:
                return i

            leftSum += nums[i]

        return -1


    # Solution
    def Solution(self, nums: List[int]) -> int:
        # Initialize leftSum & rightSum to store
        # the sum of all the numbers strictly to
        # the index's left & right respectively
        leftSum, rightSum = 0, sum(nums)

        # Traverse elements through the loop
        for idx, ele in enumerate(nums):
            rightSum -= ele

            # If the sum of all the numbers strictly to
            # the left of the index is equal to the sum of
            # all the numbers strictly to the index's right
            if leftSum == rightSum:
                # Return the pivot index
                return idx

            leftSum += ele

        # If there is no index that satisfies
        # the conditions in the problem statement
        return -1
